tests/ui/try-trait/try-after-unwrap-issue-127345.rs

```diff
@@ -0,0 +1,62 @@
+fn foo() -> Option<usize> {
+    let x = Some(42).expect("moop")?;
+    //~^ ERROR the `?` operator can only be applied to values that implement `Try`
+    //~| HELP the trait `Try` is not implemented for `{integer}`
+    //~| HELP remove the `.expect()`
+    let x = Some(42).unwrap()?;
+    //~^ ERROR the `?` operator can only be applied to values that implement `Try`
+    //~| HELP the trait `Try` is not implemented for `{integer}`
+    //~| HELP remove the `.unwrap()`
+    x
+}
+
+fn bar() -> Option<usize> {
+    foo().or(Some(43)).unwrap()?
+    //~^ ERROR the `?` operator can only be applied to values that implement `Try`
+    //~| HELP the trait `Try` is not implemented for `usize`
+    //~| HELP remove the `.unwrap()`
+}
+
+fn baz() -> Result<usize, ()> {
+    Ok(44).unwrap()?
+    //~^ ERROR the `?` operator can only be applied to values that implement `Try`
+    //~| HELP the trait `Try` is not implemented for `{integer}`
+    //~| HELP remove the `.unwrap()`
+}
+
+fn baz2() -> Result<String, ()> {
+    Ok(44).unwrap()?
+    //~^ ERROR the `?` operator can only be applied to values that implement `Try`
+    //~| HELP the trait `Try` is not implemented for `{integer}`
+    //~| HELP remove the `.unwrap()`
+}
+
+fn baz3() -> Option<usize> {
+    Ok(44).unwrap()?
+    //~^ ERROR the `?` operator can only be applied to values that implement `Try`
+    //~| HELP the trait `Try` is not implemented for `{integer}`
+    //~| HELP remove the `.unwrap()`
+}
```

Comment:
In `baz3` the suggestion shouldn't fire because removing the `.unwrap` does not help. You probably want to check that the return type of the function is compatible with the non-`unwrap`ped reciever